### DISCUSSION TITLE
Contribution percentage by all independent variables

### DISCUSSION THREAD

**Discussion by AruunnAnanda** [google/lightweight_mmm#119] [Ideas]
Hi there,

It would be great if we had a plot function that would create contribution percentage of all independent variables. There are companies whose revenue is largely influenced by **non-media channels** and not just by media spend.

As of now I believe there is only provision for media channels. I'm sharing a screenshot from Robyn MMM output where the graph contains contribution percentage of all independent variables through which we get a comprehensive overview in a glance. 

Thank you!

![image](https://user-images.githubusercontent.com/107944288/207082241-a3855417-cc8b-4162-8a56-fb158d28a5b8.png)

**Comment by statm3n** [comment]
hi @AruunnAnanda 

Thanks for sharing your feedback! 

Waterfall plot is within our radar and has been evaluated! 

We have been working on a few new plots for both pre and post model analysis. Please be on the watch out for them in future commits!

**Comment by AruunnAnanda** [comment]
Hi @statm3n ,

Okay, thank you! Is there a way to find out the contribution percentage from non-media variables now?

**Reply by statm3n** [reply]
Not explicitly. Additional engineering is needed to get non-media contributions out of the fitted mmm object. We have been working on a few new features including waterfall plots. 

In the interim, feel free to unwrap the fitted mmm object. You could refer to https://github.com/google/lightweight_mmm/blob/main/lightweight_mmm/plot.py line 143 create_media_baseline_contribution_df regarding creating non-media contributions. @AruunnAnanda

**Reply by AruunnAnanda** [reply]
Okay, thanks!

Do you have any tentative date in mind about when the new update to the package could happen?
